examples/step-15/doc/results.dox

```diff
@@ -204,3 +204,47 @@ iterations.
 
 How to do this is, again, not entirely trivial, and we again leave it as a
 future exercise.
+
+
+
+<h4> Using automatic or symbolic differentiation to compute the Jacobian matrix </h4>
```

Comment:
Since we only have the devices to use AD to directly compute the Jacobian matrix, I would suggest that we limit the discussion to that. My experience has been that SD is too computationally intensive to directly linearise the residual, and that we'd need to use another framework to employ it at the finite element level (this is what I've done in my "weak forms" side project). I'd be concerned that mentioning SD here might lead someone to try to use it for residual linearisation, which might work for this problem but would be intractable for problems with more DoFs or sophisticated constitutive relations.